flink-table/flink-table-api-java/src/test/java/org/apache/flink/table/test/program/SinkTestStep.java

```diff
@@ -37,6 +37,10 @@ public final class SinkTestStep extends TableTestStep {
     public final @Nullable List<Row> expectedAfterRestore;
     public final @Nullable List<String> expectedBeforeRestoreStrings;
     public final @Nullable List<String> expectedAfterRestoreStrings;
+    // These are added for situations where we need to specify the output in batch.
+    // In many cases, the "expectBeforeRestore*" variables are sufficient.
+    public final @Nullable List<Row> expectedBatchRows;
```

Comment:
Call this `expectedMaterializedRows` and `expectedMaterializedStrings`. Batch is just an execution mode. In theory, we can also use those two fields for streaming mode.